src/security/decorators.py

```diff
@@ -1007,6 +1007,9 @@ def wrapper(request, *args, **kwargs):
         if request.user in preprint.subject_editors():
             return func(request, *args, **kwargs)
 
+        if request.user in request.repository.managers.all():
```

Comment:
I think would be nice to use the function you just added above:
`if request.user.is_staff or request.user.is_repository_manager(request.repository)`